include/mrdocs/ConfigOptions.inc

```diff
@@ -69,6 +69,8 @@ COMMON_OPTION (compilationDatabase     , compilation-database    , Path to the c
 // Build options
 COMMON_OPTION (cmake                   , cmake                   , CMake arguments when generating the compilation database from CMakeLists.txt)
 COMMON_OPTION (defines                 , defines                 , Additional defines passed to the compiler)
+COMMON_OPTION (useSystemStdLib         , use-system-stdlib       , True if the compiler has to use just the system standard library)
+COMMON_OPTION (stdLibPaths             , stdlib-path             , Standard Library include paths)
```

Comment:
I'm not sure I fully understand what you're proposing, however...
This PR already incorporates `-isystem` with the `-stdlib-path` option.

To maintain focus, I propose that this PR remains centered exclusively on resolving issue #644 with the current implementation.
We can explore the addition of `-I` in a future PR, discussing its integration and benefits in more depth.
What do you think?

Reply:
> This PR already incorporates -isystem with the -stdlib-path option.

Yes. But it's the other way around. This PR should already incorporates -stdlib-path with the -isystem option. Because -isystem is the general case. Not -stdlib-path.

Let's just this new options `stdlib-paths` make it represent `-isystem` instead of `-stdlib` because, in our case, 1) `-stdlib` is just a subset of `-isystem` (we only need the headers in MrDocs - there's no linking), and 2) the syntax of `-stdlib` is `-stdlib=<library>` but that's not our syntax, which is confusing.

In other words, let's rename this. We already have `defines` for what is `PREDEFINED` in Doxygen. Let's create an `includes` option, which we can apply in the database as `-I`, and another option we can apply in the database as `-isystem`, so we can call it `system-includes`.

So instead of `stdlib-paths`, we will have `includes` and `system-includes`. Instead of `stdlib-paths`, we will use `system-includes` for these stdlib paths. This way, we already implement non-ambiguous options we'll need anyway and already solve the problem about whether `stdlib-paths` should be public or not.

Reply:
Thank you for clarifying that. I believe the main point of confusion here is that what I implemented as `-stdlib-path` is actually a search path for standard library header files, which in GCC and Clang is equivalent to using `-isystem`. So, it's essentially the same functionality, just under a different name.

I will add support for `-includes` (similar to `-I` in GCC terminology) and will rename `-stdlib-paths` to `-system-includes` ...

Reply:
@alandefreitas 

I've recently encountered some compilation issues in my PR, which you can see here:

https://github.com/cppalliance/mrdocs/actions/runs/10212908784/job/28257619626#step:10:64

I managed to resolve these issues by modifying how we include LibC++:

`new_cmdline.emplace_back(fmt::format("-stdlib++-isystem{}", inc));`

Instead of using:

`new_cmdline.emplace_back(fmt::format("-isystem{}", inc));`

Using the latter `-isystem` results in the errors linked above.

Given that in what we now call `system-includes` we do not distinguish between C++ and C libraries, and our default `system-includes` already include LibC++ (for C++) and Clang includes (for C), I am thinking that we might need to differentiate between C and C++:

1. `includes`: include path for header files from user libraries, applicable to both C and C++ (indistinct here).
2. `system-includes`: include path for system library header files for the C language.
3. `system-includes-cxx`: include path for system library header files for the C++ language.

What are your thoughts?

Reply:
Yes. The headers in `/usr/local/share/mrdocs/libcxx/...` end up including the headers in `/usr/include`. Should this be allowed to happen at all? 

Looking at https://libcxx.llvm.org/UsingLibcxx.html, did you give clang the options to remove these standard paths? Like `-nostdinc++` and `-nostdlib++`.

I also don't see that `-stdlib++-isystem` option documented anywhere. Where did you find it? What is it supposed to do that `-nostdinc++`, `-nostdlib++`, and `-isystem` can't?

>  I am thinking that we might need to differentiate between C and C++

Why? MrDocs has no linking anyway. Even C and C++ compilers are often symlinks to the same executables.

> system-includes-cxx: include path for system library header files for the C++ language

maybe `system-includes-libcxx`, but I'm still failing to see what this translates to

Reply:
> Yes. The headers in `/usr/local/share/mrdocs/libcxx/...` end up including the headers in `/usr/include`. Should this be allowed to happen at all?

That is peculiar. Although it is not well-documented, I believe this is due to the difference between `-stdlib++-isystem` and `-isystem`.

> Looking at https://libcxx.llvm.org/UsingLibcxx.html, did you give clang the options to remove these standard paths? Like `-nostdinc++` and `-nostdlib++`.

Yes, I did that in the past, trying various combinations until I found one that works, which is the one in the current code.

> I also don't see that `-stdlib++-isystem` option documented anywhere. Where did you find it? What is it supposed to do that `-nostdinc++`, `-nostdlib++`, and `-isystem` can't?

That option is listed here in this link:

https://clang.llvm.org/docs/ClangCommandLineReference.html

It does not explain the difference between `-isystem` and `-stdlib++-isystem` but it is designed for the C++ standard library according to the documentation: "Use directory as the C++ standard library include path". That's why I am proposing we add a similar option in MrDocs, to differentiate C++ StdLib from other includes.

> Why? MrDocs has no linking anyway. Even C and C++ compilers are often symlinks to the same executables.

What I am proposing is not related to linking. I believe Clang treats the inclusion of the C++ standard library specially, and this might be related to `#include_next`. I think if Clang uses headers that are in the directory specified by `-stdlib++-isystem`, it stops searching even if there is an `#include_next`.

> maybe `system-includes-libcxx`, but I'm still failing to see what this translates to

I suggested `system-includes-cxx` rather than `system-includes-libcxx` because `-stdlib++-isystem` is used to specify the include path for any C++ standard library implementation, not just LibC++.

Reply:
> Although it is not well-documented

I think this is documented. https://libcxx.llvm.org/UsingLibcxx.html says you need to use -nostdinc++ -nostdlib++ when using a custom std library. So, I assume it will try to include the usual system paths otherwise (like it did in CI). If it relies on system includes, we lose the reproducibility this PR intends to achieve.

> I did that in the past, trying various combinations until I found one that works

But if we have a problem above where it includes /usr/include, then it's more like it didn't work because the PR didn't achieve its goal. It just worked in the sense that it's not using the mrdocs packaged std library. My intuition is this PR without -nostdinc++/nostdlib++ has to be wrong, because we're now depending on system headers that are not reproducible and the problem remains.

> It does not explain the difference between -isystem and -stdlib++-isystem

Yes. The explanation is unclear. Even -cxx-isystem<directory> makes more sense because we know it would skip the option in C files. But -stdlib++-isystem is quite confusing. Anyway, as long as it is a distinct option for clang, it should be OK to support it. Just keep these links so we can reference them in the documentation. It would be good to have more information on the option, so we know what we're doing, though.